extension-manifest-v3/src/background/reporting.js

```diff
@@ -216,24 +218,28 @@ const reporting = new Reporting({
   communication,
 });
 
-observe('terms', (terms) => {
-  if (terms) {
-    reporting.init().catch((e) => {
-      console.warn(
-        'Failed to initialize reporting. Leaving the module disabled and continue.',
-        e,
-      );
-    });
-  } else {
-    reporting.unload();
-  }
-});
+const setup = asyncSetup([
+  observe('terms', (terms) => {
+    if (terms) {
+      reporting.init().catch((e) => {
+        console.warn(
+          'Failed to initialize reporting. Leaving the module disabled and continue.',
+          e,
+        );
+      });
+    } else {
+      reporting.unload();
+    }
+  }),
+]);
 
 function delay(timeInMs) {
   return new Promise((resolve) => setTimeout(resolve, timeInMs));
 }
 
-function onLocationChange(details) {
+async function onLocationChange(details) {
+  setup.pending && (await setup.pending);
```

Comment:
For clearer error messages (avoiding unhandled rejections if startup timed out or failed):

```
try {
  setup.pending && (await setup.pending);
} catch (e) {
  console.warn('reporting is unavailable:', e);
  return;
}
```